macos/tasks/main.yml

```diff
@@ -0,0 +1,74 @@
+---
+
+# Tasks to be run to provision the macOS buildfarm
+
+# Homebrew stuff
+
+# We redirect stdin from /dev/null so it won't prompt for install.
+#- 
+#  name: "Installing Homebrew"
+#  shell: ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)" < /dev/null
+# NOTE: Might be best to use https://github.com/geerlingguy/ansible-role-homebrew to avoid prompts etc.
+
+
+- 
+  name: "Installing Homebrew packages"
+  homebrew: 
+    name: "{{ item.name }}"
+  with_items: "{{ brew_packages }}"
+
+# Ruby stuff
+
+- 
+  name: "Install RVM prerequisites"
+  # Use get_url here
+  command: curl -sSL "https://rvm.io/mpapis.asc | gpg --import -"
+
+-
+  name: "Retrieve RVM install script"
+  get_url: 
+    url: https://get.rvm.io
+    dest: "{{tmp_path}}/install-rvm.sh"
+  register: rvm_prerequisites
+
+-
+  name: "Installing RVM"
+  command:  "bash {{tmp_path}}/install-rvm.sh stable"
+
+- 
+  name: "Install Ruby"
+  # Probably need to source ~/.bash_profile to just use rvm
+  command: "$HOME/.rvm/bin/rvm install latest-2.3"
+
+- 
+  name: "Install Gem packages"
+  gem:
+    name: "{{ item.name }}"
+    version: "{{ item.version | default(omit) }}"
+  with_items: "{{ gem_packages }}"
+
+# Node stuff
+
+-
+  name: "Retrieve NPM prerequisites"
+  get_url: 
+    url: https://raw.githubusercontent.com/creationix/nvm/v0.33.2/install.sh
+    dest: "{{tmp_path}}/install-npm.sh"
+
+- 
+  name: "Installing NVM"
+  command: "bash {{tmp_path}}/install-npm.sh"
+
+-
+  name: "Install Node"
+  # Something is dodgy here.
+  command: "nvm install {{ item }}"
+  with_items: "{{ node_versions }}"
+
+- 
+  name: "Install NPM packages"
+  npm:
+    name: "{{ item.name }}"
+    version: "{{ item.name | default(omit) }}"
+    global: yes
```

Comment:
I think item.name is correct for the way you have the npm list currently defined but it would make sense to remove the nesting from that list anyway 👍